Sming/build.mk

```diff
@@ -84,6 +84,8 @@ GIT ?= git
 # CMake command
 CMAKE ?= cmake
 
+CLANGFORMAT ?= clang-format
```

Comment:
I think `CLANG_FORMAT` would be more consistent. Also, add `DEBUG_VARS += CLANG_FORMAT`.

Reply:
done.
I noticed that cmake and git are not in DEBUG_VARS. Should I add them, too?

Reply:
Think I left those out originally as it seemed unlikely they'd be changed, but I can't think of any reason not to add them as well. I've tried to stick with the convention of 'declaring' the variables before actually assigning them, like this:

```
# Git command
DEBUG_VARS += GIT
GIT ?= git

# CMake command
DEBUG_VARS += CMAKE
CMAKE ?= cmake

# clang-format command
DEBUG_VARS	+= CLANG_FORMAT
CLANG_FORMAT ?= clang-format
```

Reply:
> Think I left those out originally as it seemed unlikely they'd be changed

I would have thought so, too, but one can never know. 
Changed it according to your suggestion.